File: labs/lab5/C#/RiadkyLib.cs
using System;

namespace RiadkyLib {
    public abstract class Riadky {
        protected string str;
        public Riadky(string s) {
            str = s;
        }
        public abstract int GetLength();
        public abstract string InsertChar();
    }

    public class VelykiLitery : Riadky {
        public VelykiLitery(string s) : base(s) { }
        public override int GetLength() => str.Length;
        public override string InsertChar() {
            string res = "";
            foreach (char c in str) {
                res += c;
                res += '/';
            }
            return res;
        }
    }

    public class MaliLitery : Riadky {
        public MaliLitery(string s) : base(s) { }
        public override int GetLength() => str.Length;
        public override string InsertChar() {
            string res = "";
            foreach (char c in str) {
                res += c;
                res += '\\';
            }
            return res;
        }
    }
}

File: labs/lab5/C++/RiadkyLib.cpp
#include "RiadkyLib.h"

// реалізація базового
Riadky::Riadky(const std::string& s) : str(s) {}
Riadky::~Riadky() {}

// VelykiLitery: після кожного символу вставляємо '/'
VelykiLitery::VelykiLitery(const std::string& s) : Riadky(s) {}
int VelykiLitery::getLength() const {
    return static_cast<int>(str.length());
}
std::string VelykiLitery::insertChar() const {
    std::string res;
    for (char c : str) {
        res += c;
        res += '/';
    }
    return res;
}

// MaliLitery: після кожного символу вставляємо '\'
MaliLitery::MaliLitery(const std::string& s) : Riadky(s) {}
int MaliLitery::getLength() const {
    return static_cast<int>(str.length());
}
std::string MaliLitery::insertChar() const {
    std::string res;
    for (char c : str) {
        res += c;
        res += '\\';  // '\\' – це один символ '\'
    }
    return res;
}
